src/components/Subscriber/Subscriber.tsx

```diff
@@ -0,0 +1,100 @@
+import Input from '@/components/Input';
+import MultiSelect from '@/components/MultiSelect';
+import Button from '@/components/Button';
+import { getTags } from '@/apis/tag';
+import { useEffect, useState } from 'react';
+import { Formik, Form } from 'formik';
+import { subscriberValidationSchema } from '@/validations/subscriber';
+
+const Subscriber = ({
+  onSubmit,
+  selectedOptions,
+  setSelectedOptions,
+  subscriber = null,
+}: {
+  onSubmit: any;
+  selectedOptions: any;
+  setSelectedOptions: any;
+  subscriber: any;
+}) => {
+  const [tags, setTags] = useState([]);
+
+  useEffect(() => {
+    const fetchTagLists = async () => {
+      setTags(await getTags());
+    };
+
+    fetchTagLists();
+  }, []);
+
+  const tagsList =
+    Array.isArray(tags.data) && tags.data.length > 0
+      ? tags.data.map((tag: any) => ({
+          value: tag.id,
+          label: tag.name,
+        }))
+      : [];
+
+  const initialValues = {
+    email: subscriber?.email ? subscriber.email : '',
+    name: subscriber?.name ? subscriber.name : '',
+    phone: subscriber?.phone ? subscriber.phone : '',
+  };
+
+  function handleTagSelection(data: any) {
+    if (data.length > 0) {
```

Comment:
if data length is zero, that means it's an empty array right? So `setSelectedOptions(data)` it will still be equal to the else condition. Why do we need this check then?

Reply:
Yes, It was a problem with the backend, I had updated it in the backend but not updated with the frontend. now I fixed